docs/Building.md

```diff
@@ -1,28 +1,173 @@
-# Building Binaries
+# Building and testing Calico Docker
 
-You will require 2 binary builds to use the Calico Docker Prototype:
+This document describes how to build the Calico Docker binaries and how to run the test suites.
+
+
+## Building Binaries and running the test suites
+
+You will require 2 binary builds to use Calico Docker:
  1. The `calico/node` Docker image which contains the Calico services.
  2. The `calicoctl` control binary to control your Calico-enabled cluster from the CLI.
 
-## Building the Docker Image
+The following sections describe how to build the binaries, and how to run the Unit Test
+and System Test suites.
 
- From the root directory of the checked out repository, run
+### Building the 'calico/node' Docker Image
+
+From the root directory of the checked out repository, run
 
     sudo make node
 
 This builds the Dockerfile giving it the name `calico/node:latest`, which is the default image name `calicoctl` will look for when starting Calico services.
 
-## Building `calicoctl` CLI Tool - mainline
+### Building `calicoctl` CLI Tool
+
+Our mainline makefile target uses a Debian-based container to build the `calicoctl` tool.  If this is not appropriate
+for your Linux distribution, you can build the tool in a python virtualenv on your distro.  Follow the appropriate
+instructions below.
+
+#### Mainline build (Debian-based container build)
+
+From the root directory of the checked out repository, run
 
     sudo make binary
 
 This builds `calicoctl` command line tool in the `dist/` directory.
 
-## Building `calicoctl` CLI Tool - virtualenv
+#### Virtual environment build
+
+Setup a virtualenv using the instructions detailed [here](#setting-up-a-virtualenv).
```

Comment:
Rather than linking with a generic “here”, it would be better to give the link some descriptive text like “using the instructions below”.